hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkTable.java

```diff
@@ -66,4 +77,34 @@ protected HoodieSparkTable(HoodieWriteConfig config, HoodieEngineContext context
   protected HoodieIndex<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>, JavaRDD<WriteStatus>> getIndex(HoodieWriteConfig config, HoodieEngineContext context) {
     return SparkHoodieIndex.createIndex(config);
   }
+
+  /**
+   * Fetch instance of {@link HoodieTableMetadataWriter}.
+   *
+   * @return instance of {@link HoodieTableMetadataWriter}
+   */
+  @Override
+  public Option<HoodieTableMetadataWriter> getMetadataWriter() {
+    synchronized (this) {
+      if (!isMetadataInfoUpdated.getAndSet(true)) {
+        // this code assumes that bootstrap of metadata table is done elsewhere (SparkRDDWriteClient) and so, we gauge whether metadata table
+        // is available or not once here and reuse the same info for repeated calls to getMetadataWriter(). Please remove memoization if
+        // that's not the case.
+        try {
+          if (!config.isMetadataTableEnabled() || !metaClient.getFs().exists(new Path(HoodieTableMetadata.getMetadataTableBasePath(metaClient.getBasePath())))) {
+            isMetadataTableAvailable = false;
+          } else {
+            isMetadataTableAvailable = true;
+          }
+        } catch (IOException e) {
+          throw new HoodieMetadataException("Could not instantiate metadata table writer", e);
+        }
+      }
+    }
+    if (isMetadataTableAvailable) {
```

Comment:
`Option.of(isMetadataTableAvailable).filter(p -> p).map(p -> SparkHoodieBackedTableMetadataWriter.create(context.getHadoopConf().get(), config, context))`?

Reply:
something without the if . (general code style issue, that I wonder a lot abt to get rid of these if blocks with option).

Reply:
isMetadataTableAvailable is not an Optional element. it is just a boolean. and so I could not do Option.of(boolean).filter()... I will keep trying out few things here if I can get rid of if else. for now, will leave it as is.  I could make isMetadataTableAvailable an Option<boolean> and set it explicitly within the sync block only if its avaialble and leave it as Option.empty if metadata is not available, but feel its not neat either.